src/test/java/teammates/test/cases/action/BaseActionTest.java

```diff
@@ -711,4 +711,19 @@ protected void verifyNumberOfEmailsSent(Action action, int emailCount) {
         assertEquals(emailCount, action.getEmailSender().getEmailsSent().size());
     }
 
+    protected String getPageResultDestination(String parentUri, boolean isError, String userId) {
+        String pageDestination = parentUri;
+        pageDestination = addParamToUrl(pageDestination, Const.ParamsNames.ERROR, Boolean.toString(isError));
+        pageDestination = addParamToUrl(pageDestination, Const.ParamsNames.USER_ID, userId);
+        return pageDestination;
+    }
+
+    protected static String addParamToUrl(String url, String key, String value) {
+        if (key == null || key.isEmpty() || value == null || value.isEmpty()
+                || url.contains("?" + key + "=") || url.contains("&" + key + "=")) {
+            return url;
+        }
+        return url + (url.contains("?") ? "&" : "?") + key + "=" + value;
+    }
```

Comment:
Yes there is a similar method in Url class but there sanitization is used due to there were issues for various character to append in Url like it was converting % to %25 and + to %2B.
So @leeyimin suggested to write similar method in here